lib/puppet/type/yumrepo.rb

```diff
@@ -304,19 +304,25 @@
 
   newproperty(:proxy) do
     desc "URL of a proxy server that Yum should use when accessing this repository.
-      This attribute can also be set to `'_none_'`, which will make Yum bypass any
-      global proxy settings when accessing this repository.
+      This attribute can also be set to '_none_' (or '' for EL >= 8 only),
+      which will make Yum bypass any global proxy settings when accessing this repository.
       #{ABSENT_DOC}"
 
     newvalues(%r{.*}, :absent)
     validate do |value|
       next if value.to_s =~ %r{^(absent|_none_)$}
-      parsed = URI.parse(value)
+      next if value.to_s.empty? && Facter.value(:operatingsystemmajrelease).to_i >= 8
```

Comment:
Was this tested on fedora, as I expect operatingsystemmajrelease there to be 28

Reply:
this is linked to the `libdnf` version on the box. the fix to add `_none_` as a backwards compatible option in libdnf: https://github.com/rpm-software-management/libdnf/commit/ea749e6238cdb54fe3b5add67875ab5912d2e766

Fedora has always used DNF, so `_none_` shouldn't have worked there also. I think we should accept `''` for all Fedora releases.